lib/web/apiserver.go

```diff
@@ -1751,6 +1751,28 @@ func (h *Handler) changeUserAuthentication(w http.ResponseWriter, r *http.Reques
 		return nil, trace.Wrap(err)
 	}
 
+	if h.ClusterFeatures.GetCloud() {
+		authPreference, err := ctx.clt.GetAuthPreference(r.Context())
+
+		if err == nil && authPreference.GetConnectorName() != constants.PasswordlessConnector {
+			users, err := h.cfg.ProxyClient.GetUsers(false)
+
+			if err == nil && len(users) == 1 {
+				authPreference.SetConnectorName(constants.PasswordlessConnector)
+
+				err = ctx.clt.SetAuthPreference(r.Context(), authPreference)
+
+				if err != nil {
+					h.log.WithError(err).Warn("Failed to update auth preference.")
+				}
+			} else {
+				h.log.WithError(err).Warn("Failed to fetch users.")
+			}
+		} else {
+			h.log.WithError(err).Warn("Failed to retrieve auth preference.")
+		}
+	}
```

Comment:
This needs to be refactored to be easier to follow. See https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow

Reply:
I have refactored code to avoid nesting let me know what you think https://github.com/gravitational/teleport/pull/17834/commits/8610483085c1003adb19c04922eaed401d20aa99